packages/block-library/src/legacy-widget/edit/create-control.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+export default function createControl( {
+	id,
+	idBase,
+	instance,
+	onChangeInstance,
+	onChangeHasPreview,
+	onError,
+} ) {
+	let control, form, content, hasPreview;
+
+	function initDOM() {
+		const number = ++lastNumber;
+
+		control = document.createElement( 'div' );
```

Comment:
I have no issues with all these `createElement` boilerplate, as they probably won't have impact on performance after gzip. But for readability sake, what do you think if we create a `createElement` helper function for these 😆. Could look like something like this.

```js
function createElement(tag, attributes = {}, children = []) {
  const element = document.createElement(tag);

  for (const [attribute, value] of Object.entries(attributes)) {
    element.setAttribute(attribute, value);
  }

  for (const child of children) {
    element.appendChild(child);
  }
}

const control = createElement('div', { class: 'widget open' }, [
  createElement('form', { class: 'widget-inside', method: 'post' }, ...),
]);
```

Reply:
I umm'd and ahh'd about doing this. It's more readable but you force the reader to learn something new. I'll try it and see if I like it 😀

Reply:
We could go a step further and use jsx to simplify all these:

```jsx
/** @jsx createElement */

const control = <div className="widget open">
  <form className="widget-inside" method="post">
    // ...
  </form>
</div>
```

The readability improves but now it might be confused with React 😅 .

Another idea would be using [`htm`](https://www.npmjs.com/package/htm). It's under 600 bytes so I think it's totally acceptable.

```js
const html = htm(createElement);

const control = html`
  <div class="widget open">
    <form class="widget-inside" method="post">
      // ...
    </form>
  </div>
`;
```

Reply:
That looks really nice but I'm not sure about adding `htm` as a dependency to `block-library` when only one block needs it.

I don't hate the `createElement` thing: ff40bb4d45. What do you think?

Reply:
Looking great! 👍